source/dyaml/constructor.d

```diff
@@ -540,21 +540,21 @@ unittest
 /// Construct a binary (base64) _node.
 ubyte[] constructBinary(ref Node node)
 {
+    import std.ascii : newline;
+    import std.array : array;
+
     string value = node.as!string;
     // For an unknown reason, this must be nested to work (compiler bug?).
     try
     {
-        try{return Base64.decode(value.removechars("\n"));}
-        catch(Exception e)
-        {
-            throw new Exception("Unable to decode base64 value: " ~ e.msg);
-        }
+        return Base64.decode(value.representation.filter!(c => !newline.canFind(c)).array);
```

Comment:
As it's converted to `ubyte[]` anyways, we can already do it here and save a bit on the performance side.